R/tt_pos_and_access.R

```diff
@@ -939,42 +1025,62 @@ setMethod("[", c("VTableTree", "numeric", "numeric"),
           function(x, i, j, ..., drop = FALSE) {
     ## have to do it this way because we can't add an argument since we don't
     ## own the generic declaration
-    keep_topleft <- list(...)[["keep_topleft"]] ## returns NULL if not presesnt
+    keep_topleft <- list(...)[["keep_topleft"]] ## returns NULL if not present
     keep_titles <- list(...)[["keep_titles"]] %||% FALSE
+    keep_footers <- list(...)[["keep_footers"]] %||% keep_titles
     reindex_refs <- list(...)[["reindex_refs"]] %||% TRUE
-    if(is.null(keep_topleft))
-        keep_topleft <- NA
-
+    
+    # browser()
     nr <- nrow(x)
     nc <- ncol(x)
     i <- .j_to_posj(i, nr)
     j <- .j_to_posj(j, nc)
 
     ##  if(!missing(i) && length(i) < nr) {
-    if(length(i) < nr) { ## already populated by .j_to_posj
-        keep_topleft <- isTRUE(keep_topleft)
+    if(length(i) < nr) {## already populated by .j_to_posj
+        if (is.null(keep_topleft)) keep_topleft <- FALSE
         x <- subset_by_rownum(x, i,
                               keep_topleft = keep_topleft,
-                              keep_titles = keep_titles)
-    } else {
-        keep_topleft <- !identical(FALSE, keep_topleft)
+                              keep_titles = keep_titles,
+                              keep_footers = keep_footers)
+    } else if (is.null(keep_topleft)) {
+        keep_topleft <- TRUE
```

Comment:
Why is `NULL` value of `keep_topleft` now translating to `TRUE` when it translates to `FALSE` above??

Reply:
I thought about this and I cannot find a better solution. The feature here to maintain is that whenever only columns are subsetted, the topleft information, if not specified, is maintained. The original implementation sent down the `NA` value and then it was treated differently from the other parameters in different functions (columns or row subsetting). I think it is better to have the `NULL` or `NA` option only on the first level and assume it is a boolean in all other cases. To do so I have to keep it as "missing" until it is known if only columns are subsetted.